src/NHSD.BuyingCatalogue.Organisations.Api/Models/ApplicationUser.cs

```diff
@@ -5,18 +5,67 @@ namespace NHSD.BuyingCatalogue.Organisations.Api.Models
 {
     public sealed class ApplicationUser : IdentityUser
     {
-        public string FirstName { get; set; }
-
-        public string LastName { get; set; }
+        public string FirstName { get; }
 
+        public string LastName { get; }
+        
         public string DisplayName => $"{FirstName} {LastName}";
 
-        public Guid PrimaryOrganisationId { get; set; }
+        public Guid PrimaryOrganisationId { get; }
+        
+        public OrganisationFunction OrganisationFunction { get; }
+
+        public bool Disabled { get; private set; }
+
+        public bool CatalogueAgreementSigned { get; private set; }
+        
+        private ApplicationUser()
+        {
+        }
+
+        private ApplicationUser(
+            Guid userId,
+            string userName,
+            string firstName, 
+            string lastName, 
+            string phoneNumber, 
+            string email, 
+            OrganisationFunction organisationFunction,
+            Guid primaryOrganisationId) : this()
+        {
+            Id = userId.ToString();
+            UserName = userName?.Trim();
+            NormalizedUserName = UserName?.ToUpperInvariant();
+            FirstName = firstName?.Trim();
+            LastName = lastName?.Trim();
+            PhoneNumber = phoneNumber;
+            Email = email?.Trim();
+            NormalizedEmail = Email?.ToUpperInvariant();
+            OrganisationFunction = organisationFunction ?? throw new ArgumentNullException(nameof(organisationFunction));
+            PrimaryOrganisationId = primaryOrganisationId;
+            Disabled = false;
+            CatalogueAgreementSigned = false;
+        }
 
-        public string OrganisationFunction { get; set; }
+        public static ApplicationUser CreateBuyer(
+            string userName,
+            string firstName,
+            string lastName,
+            string phoneNumber,
+            string email,
+            Guid primaryOrganisationId)
+        {
+            return new ApplicationUser(Guid.NewGuid(), userName, firstName, lastName, phoneNumber, email, OrganisationFunction.Buyer, primaryOrganisationId);
+        }
 
-        public bool Disabled { get; set; }
+        public void MarkAsDisabled()
```

Comment:
It could just be a property but this tends to lead to conditional checks & eventually we might have business logic surrounding the settings of that property e.g. raise an event for notification purposes. By implementing methods like this it makes it easier to understand the business requirements associated with the application user. I would introduce a reenable method, this could then send an email to get the user to change their password for instance & potentially be an asynchronous method. Async properties don't work very well as they are not obvious [Async properties resource](https://blog.stephencleary.com/2013/01/async-oop-3-properties.html). OK i am trying to guestimate here but I thought the method was simple enough as well.